Sources/InfomaniakCore/Asynchronous/SendableArray.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ Infomaniak Core - iOS
+ Copyright (C) 2023 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+/// A thread safe Array wrapper that does not require `await`. Conforms to Sendable.
```

Comment:
If if does not require await this will block forward progress in async code (or block the main thread). This should only be used on a background thread.

Reply:
You are correct, and in a perfect world, no-one should use { `SendableArray`, `SendableDictionary` }

If I see a Dictionary or an array used in an environement mixed between different async mechanism, this is a drop in replacement that removes race conditions.

Reply:
Fine by me. We should just keep in mind that those thread safe access can come at a cost.

Reply:
I added a comment on both classes that this should be used as a last resort or for prototyping.